testing/tools/integration/external.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2017 The Wallaroo Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+#  implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+import logging
+import os
+import shlex
+import shutil
+import socket
+import subprocess
+
+
+def ex_validate(cmd):
```

Comment:
How about having something like
`ShellCmdResult = namedtuple('ShellCmdResult', 'success output return_code command')`
and then have a simple name like:
`run_shell_cmd(cmd) => ShellCmdResult`

Reply:
I got tripped up by the name `ex_validate` previously, as we're already using the term `validate` to describe the process of validating wallaroo app outputs.

Reply:
Good catch. There's a lot of vestigial naming in there!